front50-cassandra/src/main/groovy/com/netflix/spinnaker/front50/model/application/CassandraApplicationDAO.groovy

```diff
@@ -156,44 +144,12 @@ class CassandraApplicationDAO implements ApplicationDAO {
 
   @Override
   Set<Application> search(Map<String, String> attributes) {
-    def searchableApplications = all()
-    attributes = attributes.collect { k,v -> [k.toLowerCase(), v] }.collectEntries()
-
-    if (attributes["accounts"]) {
-      // CQL 3.0/Cassandra 1.2 doesn't support the filtering of collections (3.1/2.0+ do)
-      def accounts = attributes["accounts"].split(",").collect { it.trim().toLowerCase() }
-      searchableApplications = searchableApplications.findAll {
-        def applicationAccounts = (it.accounts ?: "").split(",").collect { it.trim().toLowerCase() }
-        return applicationAccounts.containsAll(accounts)
-      }
-
-      // remove the 'accounts' search attribute so it's not picked up again in the field-level filtering below
-      attributes.remove("accounts")
-    }
-
-    // filtering vs. querying to achieve case-insensitivity without using an additional column (small data set)
-    def items = searchableApplications.findAll { app ->
-      def result = true
-      attributes.each { k, v ->
-        if (!v) {
-          return
-        }
-        if (!app.hasProperty(k) && !app.details().containsKey(k)) {
-          result = false
-        }
-        def appVal = app.hasProperty(k) ? app[k] : app.details()[k] ?: ""
-        if (!appVal.toString().toLowerCase().contains(v.toLowerCase())) {
-          result = false
-        }
-      }
-      return result
-    } as Set
-
-    if (!items) {
-      throw new NotFoundException("No Application found for search criteria $attributes")
-    }
+    return ApplicationDAO.Searcher.search(all(), attributes);
```

Comment:
Split this logic out such that it could be similarly shared between Cassandra and S3.